src/dshould/package.d

```diff
@@ -18,21 +18,59 @@ public import dshould.thrown;
 public void equal(Should, T)(Should should, T value, Fence _ = Fence(), string file = __FILE__, size_t line = __LINE__)
 if (isInstanceOf!(ShouldType, Should))
 {
+    import dshould.prettyprint : prettyprint;
+    import std.conv : to;
     import std.json : JSONValue;
     import std.traits : Unqual;
 
     static if (is(typeof(should.got()) == string) && is(T == string) && !should.hasWord!"not")
     {
         dshould.stringcmp.equal(should, value, Fence(), file, line);
     }
-    else static if (
-        is(Unqual!T == JSONValue)
-        && is(Unqual!(typeof(should.got())) == JSONValue)
-        && !should.hasWord!"not"
-    )
+    else static if (!should.hasWord!"not")
     {
-        should.terminateChain;
-        should.got().toPrettyString.should.equal(value.toPrettyString, Fence(), file, line);
+        static if (
+            is(Unqual!T == JSONValue)
+            && is(Unqual!(typeof(should.got())) == JSONValue))
+        {
+            should.allowOnlyWords!().before!"equal (string)";
+            should.terminateChain;
+
+            auto got = should.got();
+            const gotString = got.toPrettyString;
+            const valueString = value.toPrettyString;
+        }
+        else static if (
+            __traits(compiles, T.init.toString())
+            && __traits(compiles, typeof(should.got()).init.toString()))
+        {
+            should.allowOnlyWords!().before!"equal (string)";
+            should.terminateChain;
+
+            auto got = should.got();
+            const gotString = got.toString().prettyprint;
+            const valueString = value.toString().prettyprint;
+        }
+        else static if (
+            __traits(compiles, T.init[0].toString())
+            && __traits(compiles, typeof(should.got()).init[0].toString()))
+        {
+            should.allowOnlyWords!().before!"equal (string)";
+            should.terminateChain;
+
+            auto got = should.got();
+            const gotString = got.to!string.prettyprint;
```

Comment:
In the case of an unmatched parenthesis, the parse tree will fail to be constructed and we will fall back to ordinary string comparison.

Reply:
What if "(" appears in one array element and ")" appears in some later array element? Instead of falling back to the original strings for the two array elements, a strange formatting will result?